llvm/unittests/SandboxIR/SandboxIRTest.cpp

```diff
@@ -4188,6 +4188,133 @@ define void @foo(i32 %arg) {
   EXPECT_EQ(NewPHI->getNumIncomingValues(), PHI->getNumIncomingValues());
 }
 
+void checkSwapOperands(sandboxir::Context &Ctx, llvm::sandboxir::CmpInst *SBCmp,
+                       llvm::CmpInst *LLVMCmp) {
+  auto OrigOp0 = SBCmp->getOperand(0);
+  auto OrigOp1 = SBCmp->getOperand(1);
+  EXPECT_EQ(Ctx.getValue(LLVMCmp->getOperand(0)), OrigOp0);
+  EXPECT_EQ(Ctx.getValue(LLVMCmp->getOperand(1)), OrigOp1);
+  // This checks the dispatch mechanism in CmpInst, as well as
+  // the specific implementations.
+  SBCmp->swapOperands();
+  EXPECT_NE(Ctx.getValue(LLVMCmp->getOperand(0)), OrigOp0);
+  EXPECT_NE(Ctx.getValue(LLVMCmp->getOperand(1)), OrigOp1);
+  EXPECT_EQ(Ctx.getValue(LLVMCmp->getOperand(1)), OrigOp0);
+  EXPECT_EQ(Ctx.getValue(LLVMCmp->getOperand(0)), OrigOp1);
```

Comment:
Hmm why do we need both `EXPECT_NE` and `EXPECT_EQ` ? Shouldn't we keep only the `_EQ` ones?